### PR TITLE
Converted sample and test for SC 3.13, improved some labels

### PR DESCRIPTION
**Pull request by FlisherOfatale**
Functionality for Star Citizen
* Trimed ifcs and toggle from label
* Manually set some mining related
* changing check from Device to actionsmaps

Test Cases
* migrated valid.xml to file to 3.13 exported format (minor change for the exported file)
* Added actionsmap.xml sample, this file is naturally added in Star Citizen now.  Format is slightly different than exported, but same parser work.  Contact Flisher for update on the guide on the website if required
* Fixed Gunfighter for NXT (easier for Flisher to debug)
* ifcs/toggle tc, adding sample from normal SC3.13
* fixing valid.xml references
* self.maxDiff = None

### CODE REVIEW

File: src/adaptors/star_citizen.py [hunk 1]
```diff
@@ -38,9 +38,11 @@ def __validate_file(self, data):
             raise Exception("File is not a valid Star Citizen XML")
         else:
             if (
-                len(parsed_xml.getElementsByTagName("devices")) == 1
```
Comment (Rexeh) [inline src/adaptors/star_citizen.py:41]: Is devices no longer a valid construct in latest version?

File: src/adaptors/star_citizen.py [hunk 2]
```diff
@@ -121,6 +123,51 @@ def add_device(self, option):
 
     def process_name(self, name):
         helper.log("Bind Name: {}".format(name), "debug")
+        # Force some Labels, this ideally need to be declared elsewhere or from an external file
+        customLabels = {
```
Comment (Rexeh) [inline src/adaptors/star_citizen.py:127]: Can we define these in outer scope at class level please? Overall format is all cool.

**Review by Rexeh — CHANGES_REQUESTED** [review]
Thanks for doing this, looks like a fairly simple change, few changes/questions if that's ok!

Is the current version incompatible with the latest version of SC? That's only thing I'd like to understand more, or whether this change now requires something specific to be done in game (vs exporting profile).

Cheers,

Rob

File: src/adaptors/star_citizen.py (hunk 1 above)
Comment (FlisherOfatale) [reply src/adaptors/star_citizen.py:41]: JD 1.4 work with the exported format of the current SC 3.13 release.
However, SC 3.13 now dump an xml automatically, but this one doesn't have the device class.

There are other minors change but it doesn't affect the parser.

File: src/adaptors/star_citizen.py (hunk 2 above)
Comment (FlisherOfatale) [reply src/adaptors/star_citizen.py:127]: I'll take alook for this, it's being redefined on every loop and it's not a good practice.

Comment (Rexeh) [reply src/adaptors/star_citizen.py:127]: Yeah they should be consts but then all good.

Comment (FlisherOfatale) [reply src/adaptors/star_citizen.py]: moved outside of the function defenition at the class level.
Will add more label shortly

File: .gitignore [hunk 3]
```diff
@@ -12,7 +12,8 @@ tests/*/__pycache__/
 .env
 temp/
 diagrams/*.svg
+diagrams/*.pdf
 play*.py
 tests/__pycache__/*.pyc
 *.ui
-*.bat
```
Comment (FlisherOfatale) [inline .gitignore:18]: Not sure why the .bat liune is flagged, because of no new line?

**Pull request merged**